thorlcr/thorutil/thormisc.cpp

```diff
@@ -332,9 +332,19 @@ class DECL_EXCEPTION CThorException : public CSimpleInterface, implements IThorE
             }
             if (slave)
             {
-                str.appendf("SLAVE #%d [", slave);
-                queryClusterGroup().queryNode(slave).endpoint().getUrlStr(str);
-                str.append("]: ");
+                if (((int)slave) < 0)
```

Comment:
Above setSlave(-sender) and setSlave(sender) might occur, how does code below know setSlave(sender) is a slave and not a node ?

Reply:
if the exception was marked as from a slave (see line 529 of thgraphslave.cpp), it will have a serialized slave #. If it does not, it will be 0. The code in thgraphmaster.cpp is testing for that 0 and setting the resulting deserialized exception with a -ve number (-sender) to denote that it's a slave.
If the deserialized slave # was >=1 then it's a slave #.